### PR TITLE
[FIX] CSS fix for kanban view and required fields

### PR DESCRIPTION
**Pull request by kujiu**
Description of the issue/feature this PR addresses:

- Fix required fields do not appear in specific style ;
- Fix ugly footer on list views ;
- Fix kanban view does not use 100% height of screen.

### CODE REVIEW

File: addons/web/static/src/less/fields_extra.less [hunk 1]
```diff
@@ -6,7 +6,9 @@
 // Required
 .o_required_modifier {
     &.o_input, .o_input {
-        background-color: #D2D2FF;
+        background-color: #D2D2FF !important; //This color does not have to be
+                                              //overwritten by grid or over view.
+        border-bottom: 2px solid @odoo-main-text-color;
```
Comment (qsm-odoo) [inline addons/web/static/src/less/fields_extra.less]: Seems like an enterprise style.

Comment (kujiu) [inline addons/web/static/src/less/fields_extra.less]: Fixed

File: addons/web/static/src/less/webclient_layout.less [hunk 2]
```diff
@@ -158,6 +158,10 @@
                     .o-position-absolute(0, 0, 0, 0);  // Get the 100% height of its flex parent
                     height: 100%;
 
+                    > div {
+                        height: 100%;
+                    }
```
Comment (qsm-odoo) [inline addons/web/static/src/less/webclient_layout.less]: I would prefer if we remove controllers' empty div... but ok ;)

File: addons/web/static/src/less/fields_extra.less [hunk 3]
```diff
@@ -6,7 +6,8 @@
 // Required
 .o_required_modifier {
     &.o_input, .o_input {
-        background-color: #D2D2FF;
+        background-color: #D2D2FF !important; //This color does not have to be
+                                              //overwritten by grid or over view.
```
Comment (qsm-odoo) [inline addons/web/static/src/less/fields_extra.less]: Just implicit conventions:
- No space before `!important`
- Space after `//`

Also I am not sure to understand the "over view" part in the comment. Maybe the whole comment is not important...

File: addons/web/static/src/less/webclient_layout.less (hunk 2 above)
Comment (kujiu) [inline addons/web/static/src/less/webclient_layout.less]: Fixed

File: addons/web/static/src/less/fields_extra.less (hunk 3 above)
Comment (kujiu) [inline addons/web/static/src/less/fields_extra.less]: Fixed

File: addons/web/static/src/less/form_view_extra.less [hunk 4]
```diff
@@ -14,6 +14,7 @@
             max-width: @sheet-max-width;
             min-height: 330px;
             margin: 0 auto;
+            display: table;
```
Comment (qsm-odoo) [inline addons/web/static/src/less/form_view_extra.less]: Never use `display: table;` ... what is this fixing ?

File: addons/web/static/src/js/views/list/list_renderer.js [hunk 5]
```diff
@@ -328,6 +328,9 @@ var ListRenderer = BasicRenderer.extend({
         if (this.hasSelectors) {
             $cells.unshift($('<td>'));
         }
+        if (!(isGrouped || this.hasSelectors)) {
+            return '';
+        }
```
Comment (qsm-odoo) [inline addons/web/static/src/js/views/list/list_renderer.js]: Can you explain this to me ? :)

File: addons/web/static/src/js/chrome/view_manager.js [hunk 6]
```diff
@@ -286,6 +286,7 @@ var ViewManager = Widget.extend(ControlPanelMixin, {
                     controller.appendTo(view.$fragment).done(function() {
                         // Remove the unnecessary outer div
                         view.$fragment = view.$fragment.contents();
+                        view.$fragment.addClass('o_view_manager_content_inner');
```
Comment (qsm-odoo) [inline addons/web/static/src/js/chrome/view_manager.js]: What I was saying before is that we will want to remove the unnecessary controller div. For now, your previous fix was OK. Adding a class on the element will just make it seems important.

File: addons/web/static/src/xml/base.xml [hunk 7]
```diff
@@ -280,7 +280,7 @@
                 <th t-if="options.deletable" class="o_list_record_delete"/>
             </tr>
         </thead>
-        <tfoot>
+        <tfoot t-if="options.selectable or aggergate_columns or options.deletable">
```
Comment (qsm-odoo) [inline addons/web/static/src/xml/base.xml]: Is it really `aggergate_columns`, with the typo ?

File: addons/web/static/src/js/views/list/list_renderer.js (hunk 5 above)
Comment (kujiu) [inline addons/web/static/src/js/views/list/list_renderer.js]: This removes empty footer on list-view.

**Pull request closed without merging**